scripts/hid_configurator/configurator_core.py

```diff
@@ -219,31 +225,44 @@ def create_fetch_report(recipient, event_id):
         a configuration value from a device.
         Recipient is a device product ID. """
 
-    assert(type(recipient) == int)
-    assert(type(event_id) == int)
+    assert type(recipient) == int
+    assert type(event_id) == int
 
     status = ConfigStatus.FETCH
     report = struct.pack('<BHBBB', REPORT_ID, recipient, event_id, status, 0)
 
-    assert(len(report) <= REPORT_SIZE)
+    assert len(report) <= REPORT_SIZE
     report += b'\0' * (REPORT_SIZE - len(report))
 
     return report
 
 
+def check_range(value, value_range):
+    return value_range[0] <= value <= value_range[1]
+
+
 def exchange_feature_report(dev, recipient, event_id, event_data, is_fetch):
     if is_fetch:
         data = create_fetch_report(recipient, event_id)
     else:
         data = create_set_report(recipient, event_id, event_data)
 
-    dev.send_feature_report(data)
+    try:
+        dev.send_feature_report(data)
+    except:
+        if not is_fetch:
+            return False
+        else:
+            return False, None
 
     for retry in range(POLL_RETRY_COUNT):
         time.sleep(POLL_INTERVAL)
 
-        response_raw = dev.get_feature_report(REPORT_ID, REPORT_SIZE)
-        response = Response.parse_response(response_raw)
+        try:
+            response_raw = dev.get_feature_report(REPORT_ID, REPORT_SIZE)
+            response = Response.parse_response(response_raw)
+        except:
+            return False
```

Comment:
better set response to None as further code will handle everything needed

Reply:
done